### PR TITLE
Refactor sharing

### PR DESCRIPTION
**Pull request by C0rby**
- extract share creation into its own method
- extract state conversion into its own method
- extract resource info retrieval
- some minor clean up
Also the responses to accepting or rejecting a share return the updated share now.

### CODE REVIEW

**Comment by C0rby** [comment]
I don't quite get why 
```
apiSharePublicLink1/deletePublicLinkShare.feature:37
apiSharePublicLink1/deletePublicLinkShare.feature:38
```
fail...
Locally they run just fine...
@individual-it or @phil-davis  do you have an idea?

Also @ishank011 or @labkode, I don't have access to the cs3org in snyk so I can't see what test is failing there. Could you tell me what I need to fix?

**Comment by C0rby** [comment]
> I don't quite get why 
> ```
> apiSharePublicLink1/deletePublicLinkShare.feature:37
> apiSharePublicLink1/deletePublicLinkShare.feature:38
> ```
> fail...

I think I found the issue.

**Comment by C0rby** [comment]
Listing a public link after renaming it fails for some reason....
Need to investigate further...

**Comment by individual-it** [comment]
sadly the logs of reva are cut off in drone

**Comment by individual-it** [comment]
I rebased on master here https://github.com/cs3org/reva/pull/1690 let's see
Next I will try to bump the oc-core commit id

**Comment by phil-davis** [comment]
> sadly the logs of reva are cut off in drone

In owncloud drone CI we put a flush of the file-system and a 30-second wait at the end of acceptance test runs. That improved the situation - pipeline step output seemed to get from the drone agent to drone server more often! I am not sure what happens currently in cs3org/reva CI.

**Comment by C0rby** [comment]
@individual-it, my refactoring is causing the issue. I just haven't found the cause yet...

**Comment by C0rby** [comment]
Arrgh I found it...

When we are doing a stat request on a share like here https://github.com/cs3org/reva/blob/28500a8e0bb49f52f176c18b26f59d598c7cf0ab/internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go#L178

we eventually end up here https://github.com/cs3org/reva/blob/28500a8e0bb49f52f176c18b26f59d598c7cf0ab/pkg/storage/fs/owncloud/owncloud.go#L746
In that line we get the file path using the OpaqueId as the key. When we rename a file, that path will be outdated because it doesn't get updated on a file rename. Because of that the stat request fails with a `not found` error.

Back in the `listPublicShares` method we check the result of the stat https://github.com/cs3org/reva/blob/28500a8e0bb49f52f176c18b26f59d598c7cf0ab/internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/public.go#L178-L179

But if you look closely you see that it doesn't check the status code of the stat response but of the listPublicShare response.

My refactoring changed that to checking the stat response and that's why it doesn't return a share after the shared file gets renamed...

**Comment by C0rby** [comment]
This is the underlying issue. https://github.com/cs3org/reva/issues/1693 Will work on it on monday

**Comment by C0rby** [comment]
Waiting for this fix to be merged: https://github.com/cs3org/reva/pull/1696

**Comment by labkode** [comment]
@ishank011 can you review this one?

**Review by butonic — APPROVED** [review]

File: internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go [hunk 1]
```diff
@@ -61,10 +61,11 @@ type Handler struct {
 	publicURL              string
 	sharePrefix            string
 	homeNamespace          string
-	resourceInfoCacheTTL   time.Duration
 	additionalInfoTemplate *template.Template
 	userIdentifierCache    *ttlcache.Cache
 	resourceInfoCache      gcache.Cache
+	resourceInfoCacheTTL   time.Duration
+	useResourceInfoCache   bool
```
Comment (ishank011) [inline internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go:68]: Should we use it in the user-configurable struct? If a user sets `useResourceInfoCache` to true but doesn't set `resourceInfoCacheTTL` assuming a default value, we won't use the cache. I'd suggest removing this and checking  the value of `resourceInfoCacheTTL` in all places

Comment (ishank011) [reply internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go:68]: Okay, no it's not user configurable. But I'd still vote for removing the extra variable. Doesn't really add any value

Comment (C0rby) [reply internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go:68]: It does add a bit of readability but I have no hard opinion on this, I can remove it again.

**Review by ishank011 — APPROVED** [review]

**Pull request merged**